sycl/include/CL/sycl/detail/common.hpp

```diff
@@ -246,6 +246,14 @@ size_t getLinearIndex(const T<Dims> &Index, const U<Dims> &Range) {
   return LinearIndex;
 }
 
+// Kernel set id, used to group kernel names together based on their
+// distribution among device images
+using KernelSetId = size_t;
```

Comment:
It is grouping kernel names into sets, just as the comment states. The relationship is many-to-one: kernel name -> kernel set id. I can't think of a way to expand this comment aside from mentioning that the sets are disjoint (applied).

Reply:
> Also I wonder why pi_device_binary object itself can't be used as a kernel set ID?

Because the device images themselves don't come into play here, rather, several device images can be associated with the same kernel/kernel set (i. e. kernel set id is mapped to a vector of device images).